### ISSUE TITLE
Setting active_layer programmatically does not change which layer is highlighted

### ISSUE THREAD

**Issue by ianhi** [napari/napari#2356]
## 🐛 Bug

Setting `viewer.active_layer` from jupyter (with `%gui qt`) only updates part of the left sidebar ui.

![Peek 2021-03-04 22-45](https://user-images.githubusercontent.com/10111092/110064799-d3823380-7d3b-11eb-9579-a0d38df78797.gif)
## To Reproduce
Run this code:

```python
import napari
import numpy as np
from skimage import data

viewer = napari.view_image(data.astronaut())
points = np.array([[100, 100], [200, 200], [300, 100]])
mask = viewer.add_labels(np.zeros([512,512]), name="mask")
viewer.add_points(points, size=30)

# set the active layer
viewer.active_layer = mask
```

## Expected behavior
The mask layer is highlighted blue in the layer list.

## Environment
```


napari: 0.4.6
Platform: Linux-5.8.0-7630-generic-x86_64-with-glibc2.10
System: Pop!_OS 20.04 LTS
Python: 3.8.5 (default, Sep 4 2020, 07:30:14) [GCC 7.3.0]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.19.5
SciPy: 1.6.0
Dask: 2021.02.0
VisPy: 0.6.6

OpenGL:
- GL version: 4.6 (Compatibility Profile) Mesa 20.2.6
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1080, scale 1.0

Plugins:
- console: 0.0.3
- svg: 0.1.4
```

**Comment by sofroniewn** [comment]
Yes, unfortunately this has never worked. We're almost certainly going to remove `active_layer`, i had one PR to do so in progress in #1952, but I think we're now going to wait until #2239 merges which adds a new selection model and then support this functionality using that feature. Sorry about this! Thanks for the very nice bug report. I'm hoping that we could get this fixed by the next release, or at least add a warning/ error message.

**Comment by ianhi** [comment]
Makes sense! 

Is there a place where how to do stuff like setting active layers is documented? If there is and you think a fix may take awhile then I can open a PR to leave a note about this.

(also if such documentation exists I'd love to read it - i'm mostly figuring this out as I go by looking at the napari code)

**Comment by tlambert03** [comment]
Unfortunately not really, napari.org is the only docs location.  It’s a work in progress... there is still a lot of functionality like this that can only be discovered by exploring the source code. This particular API is likely to change in the next couple months

**Comment by sofroniewn** [comment]
> (also if such documentation exists I'd love to read it - i'm mostly figuring this out as I go by looking at the napari code)

Maybe the best thing to do is expand the doc strings of the ViewerModel and add some comments in the code

**Comment by sofroniewn** [comment]
Will be closed by #2441
